src/bp/core/services/dialog/prompts/confirm.ts

```diff
@@ -0,0 +1,59 @@
+import { IO, Prompt, PromptConfig } from 'botpress/sdk'
+import * as sdk from 'botpress/sdk'
+import { extractEventCommonArgs } from 'common/action'
+import yn from 'yn'
+
+class PromptConfirm implements Prompt {
+  public static config: PromptConfig = {
+    type: 'confirm',
+    label: 'Confirm',
+    valueType: 'boolean',
+    minConfidence: 1,
+    noValidation: true,
+    params: {
+      question: { label: 'Question to ask to user', type: 'string' }
+    }
+  }
+
+  private _question: string
+
+  constructor({ question }) {
+    this._question = question
+  }
+
+  extraction(event: IO.IncomingEvent) {
+    const yesOrNo = yn(event.payload?.payload || event.preview)
+
+    return { value: yesOrNo, confidence: 1 }
+  }
+
+  async validate(value) {
+    return { valid: value === true || value === false, message: 'Invalid' }
+  }
+
+  // Customize the prompt event that is sent through the specific channels.
+  customPrompt = async (event: IO.OutgoingEvent, incomingEvent, bp: typeof sdk) => {
```

Comment:
I'm not a fan of having to define all the possible channel specific render within the prompts code... But I don't know, it's a tough call... on one side, we wan't prompts to be self contained, and on the other side, we want them to be channel agnostic...

Reply:
Yup, that's a catch22 issue. By default a prompt will send an event of type "prompt" with the configuration as a payload, either the channel can handle it or just ignore it. 

I've added the customPrompt so we can choose to convert that payload to something more specific, but not sure it's the best solution. Open to discussion!

Reply:
Oh, ok so what you're saying is that prompt rendering will be done in channel code and this piece of custom logic is in case something special needs to be added for some channels ? 

if that's what it is then it's not bad

Reply:
The idea is that anytime a question is asked, we ask the question to the user, and we also send an event of type "prompt" to the channel, with the configuration of the prompt as a payload. Then, in channels we may handle the different prompts, or just ignore them.

In the case of confirm, we could also implement it in channel-web and convert it to yes/no buttons (for quick replies).

Right now it doesn't send the type: 'prompt' to channels, because I haven't modified all of them to not throw an error for that type of event